internal/anchor/validator.go

```diff
@@ -81,8 +84,15 @@ func (v *Validator) handle(req *anchorRequest) admission.Response {
 
 	errStrs := validation.IsDNS1123Label(cnm)
 	if len(errStrs) != 0 {
-		msg := fmt.Sprintf("Subnamespace %s is not a valid namespace name: %s", cnm, strings.Join(errStrs, "; "))
-		return webhooks.Deny(metav1.StatusReasonInvalid, msg)
+		fldPath := field.NewPath("metadata", "name")
+		msg := fmt.Sprintf("not a valid namespace name: %s", strings.Join(errStrs, "; "))
+		allErrs := field.ErrorList{
+			field.Invalid(fldPath, cnm, msg),
+		}
+
+		gk := schema.GroupKind{Group: api.GroupVersion.Group, Kind: "SubnamespaceAnchor"}
+		err := apierrors.NewInvalid(gk, cnm, allErrs)
+		return webhooks.DenyFromStatusError(err)
```

Comment:
The term "status" (as in, `DenyFromStatusError`) seems a bit weird here. Maybe call this `DenyFromAPIError`? I know "status" is technically what the struct is called, but from the PoV of the client, it's from a package called `.../api/errors` so I think this name fits better.

Reply:
Renamed